security_check/main.py

```diff
@@ -30,12 +31,43 @@ def _run_gcloud(cmd):
     return json.loads(output)
 
 
+def _find_base_image(image):
+    parsed = _run_gcloud(['describe', image])
+    img = parsed.get('image_analysis')
+    if img:
+        base_img_url = img[0]['base_image_url']
+        return base_img_url[len('https://'):base_img_url.find('@')]
+
+
+def _check_for_vulnz(image, severity, whitelist):
+    unpatched = _check_image(image, severity, whitelist)
+    if not unpatched:
+        return unpatched
+
+    base_image = _find_base_image(image)
+    base_unpatched = {}
+    if base_image:
+        base_unpatched = _check_image(base_image, severity, whitelist)
+    for vuln in base_unpatched.keys():
+        logging.info('Vulnerability %s exists in the base image. Skipping.',
+                     vuln)
+        del unpatched[vuln]
```

Comment:
Could it be possible for the base image to contain a vuln, but not the child image? I guess if the child image explicity does an apt-get upgrade it could happen. I think in that case this line would fail.

Maybe it would be better to iterate through the unpatched list, and re-collect the vulns into a new object if they're not present in the base?